### DISCUSSION TITLE
p

### DISCUSSION THREAD

**Discussion by ghost** [smartcontractkit/full-blockchain-solidity-course-py#284] [Q&A] [answered]
o

**Comment by cromewar** [comment, marked as answer]
Hello @balabala7  please be more specific about what kind of help you need?... "and i'm trying to make it for my whole 10k collection can you help me out with this please ? " this is a very general question. I'm willing to help you, but you must ask specific questions.

Also, to maintain consistency on the discussion section please keep this question in just one thread, as you already made the exact same here: https://github.com/smartcontractkit/full-blockchain-solidity-course-py/discussions/281 . @PatrickAlphaC  could you please close the thread on the link?

**Reply by PatrickAlphaC** [reply]
Hi @balabala7, the contract you deploy is actually the collection contract. Every time you call the `mint` function, you mint a token of that specific type of collection. Does that make sense?

**Reply by PatrickAlphaC** [reply]
Not quite.

If you want 10k tokens, you call the `mint` function 10k times. Each token needs its own metadata, so you'll then need to call `setTokenURI` for each token with your tokenURI and it's metadata.

You'll put the images in something like IPFS. Please watch the NFT section learn how to do this. Or any of these three videos:

https://www.youtube.com/watch?v=p36tXHX1JD8
https://www.youtube.com/watch?v=EzZEx0lumr0
https://www.youtube.com/watch?v=9oERTH9Bkw0

**Reply by cromewar** [reply]
You would like to use `_safeMint(msg.sender, newTokenId);` function of your solidity contract, and then call that function in the fronted with React as Patrick does on the video in lesson 13: https://youtu.be/M576WGiDBdQ?t=46086
